### PR TITLE
Switch from nose to pynose

### PR DESCRIPTION
**Pull request by gherceg**
[pynose](https://pypi.org/project/pynose/) is an updated version of nose, with support for newer versions of python. The lack of support for newer versions of python was initially a reason to switch test frameworks, but pynose enables retaining the most compatibility with the current test framework while also enabling usage of newer python versions.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -60,7 +59,7 @@ jobs:
             "HOST":"localhost",
             "PORT":"5432"
           }
-      run: nosetests -v --with-coverage --cover-branches --cover-package=field_audit
+      run: pynose -v --with-coverage --cover-branches --cover-package=field_audit
```
Comment (millerdev) [inline .github/workflows/tests.yml:62]: Is this change necessary? I think `nosetests` still works with `pynose`.

**Review by millerdev — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (gherceg) [reply .github/workflows/tests.yml:62]: Yeah this is definitely true. I think I incorrectly viewed `nosetests` as the "deprecated" version but given that pynose exists simply to support nose as it is today, using `nosetests` sounds good. I'll undo this as the fewer changes the better.

File: README.md [hunk 2]
```diff
@@ -221,7 +221,7 @@ your local Python's `sqlite3` library ships with the `JSON1` extension enabled
 
 - Tests
   ```shell
-  nosetests
+  pynose
```
Comment (gherceg) [inline README.md:224]: Preference for recommending running tests with nosetests vs pynose?

Comment (millerdev) [reply README.md:224]: No strong preference either way. "The fewer changes the better"?

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]
